### ISSUE TITLE
Unhandled exception detecting "Starfield" on Linux

### ISSUE THREAD

**Issue by PickMeNow** [DoubleYouC/Bethini-Pie-Performance-INI-Editor#9]
Previous version worked well, no issues running it through wine.
I am able to select the game "Starfield" afterwards it gives Unhandled Expection window. Setting Game Path and INI Path is possible but I am not able to set or view values.

Clicking "Ok" on the unhandled exception exits the program.

**Comment by DoubleYouC** [comment]
The window should state the exception, and the log file has more information. As is, I can't tell what is happening from the description.

**Comment by PickMeNow** [comment]
Leaving a screenshot maybe it helps:

![image](https://github.com/DoubleYouC/Bethini-Pie-Performance-INI-Editor/assets/29429513/b10ac8b0-7e9b-4ec8-a414-98c0c93a1fda)

**Comment by mz1193** [comment]
Can confirm the same exact issue with Fallout 4. Rolling back to 4.8 and everything works normally.

**Comment by Ruedii** [comment]
I had this issue too.

For short term, adjust the instructions to use a tool such as proton tricks to launch the tool under proton. 

As of long term I think the best way for it to handle it is to add a launcher script for Linux users that will set it up to launch under the proton environment of the appropriate game.  You can find the core of such a linux script in the ProtonTricks project.   This script can also setup any needed runtimes that are absent from proton.  Chances are the Winetricks script ProtonTricks uses has installers for said runtimes available. 

You may also want to donate a tool script to Protontricks and/or the WineTricks tool to install and run Bethini Pie.  This would be the ideal solution.   

Still, I would like to note, this should NOT be a fatal error.  If anything goes wrong finding the directory structure, it should simply prompt the user for the appropriate directories to use.

**Comment by Continous** [comment]
> Still, I would like to note, this should NOT be a fatal error. If anything goes wrong finding the directory structure, it should simply prompt the user for the appropriate directories to use.

I agree greatly with this sentiment. Graceful failure is always better than a hard crash. Communication to the user that something has gone wrong is more than enough. Crashing and burning over a missing file or directory is typically not good.

**Comment by awebeer256** [comment]
> For short term, adjust the instructions to use a tool such as proton tricks to launch the tool under proton.

I'm seeing this issue with Skyrim SE when Bethini is launched from MO2 which replaced the Steam Skyrim SE launcher, so I'm pretty sure it's already running under Proton.

**Comment by HeyJoplin** [comment]
Hello everyone,

I've been trying to setup FO4 in a macMini M1 (Sonoma) with CrossOver 24.0.3, and got the same "Unhandled exception" error after the game selection screen. After a lot of debugging and searching, I managed to solve it. Not sure if it will solve the scenarios reported above (Linux, Starfield, etc), but I guess it's worth a try. Here it goes:

- Install Python 12 and download the latest repo code
- Install dependencies with pip. Here's what I have currently installed:
```
# pip list
Package                   Version
------------------------- --------
altgraph                  0.17.4
colorama                  0.4.6
debuglater                1.4.4
packaging                 24.1
pefile                    2023.2.7
pillow                    10.3.0
pip                       24.0
pyinstaller               6.8.0
pyinstaller-hooks-contrib 2024.7
pywin32-ctypes            0.2.2
setuptools                70.1.0
simpleeval                0.9.13
ttkbootstrap              1.10.1
```

- Edit file "Bethini.pyw", from line 807 onwards, changing from:
```
self.tab_dictionary[each_tab]["LabelFrames"][the_label_frame]["SettingFrames"][on_frame][the_setting][id_] = 
ttk.Checkbutton(self.tab_dictionary[each_tab]["LabelFrames"][the_label_frame]["SettingFrames"][on_frame][the_setting]["TkFinalSettingFrame"],
                                                                                                                              text=each_setting, variable=self.tab_dictionary[each_tab]["LabelFrames"][the_label_frame]["SettingFrames"][on_frame][the_setting]["tk_var"],
                                                                                                                              onvalue=on_value, offvalue=off_value)
```

to:

```
self.tab_dictionary[each_tab]["LabelFrames"][the_label_frame]["SettingFrames"][on_frame][the_setting][id_] = 
ttk.Checkbutton(self.tab_dictionary[each_tab]["LabelFrames"][the_label_frame]["SettingFrames"][on_frame][the_setting]["TkFinalSettingFrame"],
                                                                                                                              bootstyle="round-toggle",
                                                                                                                              text=each_setting, variable=self.tab_dictionary[each_tab]["LabelFrames"][the_label_frame]["SettingFrames"][on_frame][the_setting]["tk_var"],
                                                                                                                              onvalue=on_value, offvalue=off_value)
```
i.e. just add **bootstyle="round-toggle",** to the parameter list

- start a regular "cmd" session
- start the app with the following command: python Bethini.pyw

It will change the style of the checkboxes (or Checkbuttons, as the ttk framework calls it), but I guess it's just cosmetics.

For reference, the solution was posted in the following issue:
https://github.com/israel-dryer/ttkbootstrap/issues/506


Please let me know if it works. I'm quite rusty with Python programming, but maybe next week I can take some time to create a PR, or a clone with the new binaries.

Cheers!

**Comment by DoubleYouC** [comment]
I have already solved this issue with the help of another tester, but I haven't posted the fix yet.

**Comment by Treeways** [comment]
This bug affects all supported titles on Steam Deck and other platforms. I'm excited to use the fixed version!

**Comment by LinuxfarmerHH** [comment]
Tried 4.11 started from Vortex

couldn't open "icons\Setup.png": no such file or directory
2024-07-29 18:38:27,321 - INFO - Starfield Documents\My Games folder is Starfield.
2024-07-29 18:38:27,322 - INFO - User documents location: C:\users\eric\Documents
2024-07-29 18:38:27,373 - INFO - Starfield Documents\My Games folder is Starfield.
2024-07-29 18:38:27,374 - INFO - User documents location: C:\users\eric\Documents
2024-07-29 18:38:27,376 - INFO - No image for tab.

couldn't open "icons\Preferences.png": no such file or directory
2024-07-29 18:38:27,416 - INFO - An unhandled exception occurred.

Both image files are not part of the icons folder, will try 4.8 as recommended here

**Comment by hellomp** [comment]
For anyone still having this issue, the easy way to solve it is to install the Segoe UI font into the game prefix. Go to the repository bellow and drop all fonts from the font folder into the "/home/deck/.local/share/Steam/steamapps/compatdata/489830/pfx/drive_c/windows/Fonts" folder (I've tried other font files but it seems that it just works with a specific file name with small caps, like the ones from this repository.

https://github.com/mrbvrz/segoe-ui-linux/tree/master

**Comment by ThisNekoGuy** [comment]
Hopefully that update comes out soon so that font fix won't be required for too much longer... :/

**Comment by Rabcor** [comment]
> For anyone still having this issue, the easy way to solve it is to install the Segoe UI font into the game prefix. Go to the repository bellow and drop all fonts from the font folder into the "/home/deck/.local/share/Steam/steamapps/compatdata/489830/pfx/drive_c/windows/Fonts" folder (I've tried other font files but it seems that it just works with a specific file name with small caps, like the ones from this repository.
> 
> https://github.com/mrbvrz/segoe-ui-linux/tree/master

You only need [seguisym.ttf](https://github.com/mrbvrz/segoe-ui-linux/blob/master/font/seguisym.ttf) .

It is used for the checkmarks.
![image](https://github.com/user-attachments/assets/2a329781-8025-4403-a276-2d3b9cc20c01)

You can just copy, paste and rename one of the existing font symlinks in your prefix to seguisym.ttf, the checkmarks will look a bit funky but everything will work.

**Comment by eliduvid** [comment]
Had the same issue (with skyrim, but specific game doesn't matter) through winlator on android. Solved by just throwing [seguisym.ttf](https://github.com/mrbvrz/segoe-ui-linux/blob/master/font/seguisym.ttf) from previous comment into the BethiniPie folder. It may also work on other platforms

**Comment by DoubleYouC** [comment]
Fixed in 4.12.
